dockerfiles/theia/0.3.10/yarn.lock

```diff
@@ -0,0 +1,8015 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```

Comment:
this file became broken after each Theia release, so we store normal state of this file.

Reply:
For example after release theia 0.3.11 yarn.lock generated by yarn command contains incorrectly links to the latest theia version. And theia incorrectly start with such dependencies.

Reply:
I've understood the issue.
but what I'm asking is how we get the `yarn.lock` you're adding there
because I suppose it will be added for each theia version
like
```
dockerfiles/theia/0.3.10/yarn.lock
dockerfiles/theia/0.3.11/yarn.lock
```
so if there are some changes inside this file, it should be documented or generated automatically

Reply:
@benoitf Oh, sorry. I've got this yarn.lock from current working Theia image. But another way to get this yarn lock: run yarn command "yarn" with our current Theia package.json and will be generated yarn.lock file. If we have incorrectly yarn.lock we can edit this yarn.lock: 
For example replace all incorrectly links to correctly: 
```
"@theia/application-manager@^0.3.10":
  version "0.3.11"
  resolved "https://registry.yarnpkg.com/@theia/application-manager/-/application-manager-0.3.11.tgz#c0a284898442f7531e081426385616ead51f89a1"
  dependencies:
    "@theia/application-package" "^0.3.11"
``` 
Fixed:
```
"@theia/application-manager@^0.3.10":
  version "0.3.10"
  resolved "https://registry.yarnpkg.com/@theia/application-manager/-/application-manager-0.3.11.tgz#c0a284898442f7531e081426385616ead51f89a1"
  dependencies:
    "@theia/application-package" "^0.3.10"
``` 

Run `yarn` again and yarn detects incorrectly hash and propose correctly hash:

```
error https://registry.yarnpkg.com/@theia/application-manager/-/application-manager-0.3.10.tgz: Fetch succeeded for undefined. However, extracting "https://registry.yarnpkg.com/@theia/application-manager/-/application-manager-0.3.10.tgz" resulted in hash "657519a699a481bc1c4b62eb8b943cfad8bb17fe", which did not match the requested hash "c0a284898442f7531e081426385616ead51f89a1".
```

We can edit dependency again and use correctly hash. We can repeat this actions for all broken dependencies. And yes it's can be automated.

@benoitf Do you want that we automate manual fixing yarn.lock?

Reply:
should workaround proposed there https://github.com/theia-ide/theia/issues/1615#issuecomment-377659319 be working ?

Reply:
@benoitf,  yes https://github.com/theia-ide/theia/issues/1615#issuecomment-377659319 it works(much slower, but for docker build I don't think that it's critical). So, Do you propose, store patched package.json with resolution sections, and on creation newer theia image apply new pathched package.json?

Reply:
@AndrienkoAleksandr I would suggest that we keep the line
ADD https://raw.githubusercontent.com/theia-ide/theia/v${THEIA_VERSION}/examples/browser/package.json /home/theia/package.json

but before running yarn command on the file, we use a script to remove the carret before theia dependencies and we insert a resolutions section in the field.

So migrating from one version to another should be only a matter of updating THEIA_VERSION constant.

Reply:
Ok, variant of such fix: https://github.com/eclipse/che/pull/10164